googlesource/googlesource.go

```diff
@@ -334,79 +292,64 @@ func (s *Source) Next(ctx context.Context) (sdk.Record, error) {
+	s.position = ""
 
-	err := json.Unmarshal(pos, &s.positions.positions)
+	err := json.Unmarshal(pos, &s.position)
 	if err != nil {
 		sdk.Logger(s.ctx).Info().Msg("Could not get position. Will start with offset 0")
 	}
-	s.positions.lock.Unlock()
 }
 
 func getTables(s *Source) (err error) {
 	if s.sourceConfig.Config.TableIDs == "" {
-		s.tables, err = s.listTables(s.sourceConfig.Config.ProjectID, s.sourceConfig.Config.DatasetID)
-		if err != nil {
-			sdk.Logger(s.ctx).Trace().Str("err", err.Error()).Msg("error found while listing table")
-		}
-	} else {
-		s.tables = strings.Split(s.sourceConfig.Config.TableIDs, ",")
+		sdk.Logger(s.ctx).Trace().Str("err", err.Error()).Msg("error found while listing table")
+		return fmt.Errorf("table ID blank")
 	}
+	s.table = s.sourceConfig.Config.TableIDs
 	return err
 }
 
 func (s *Source) runIterator() (err error) {
-	rowInput := make(chan readRowInput)
+	var wg sync.WaitGroup
 
 	if err = getTables(s); err != nil {
 		sdk.Logger(s.ctx).Trace().Str("err", err.Error()).Msg("error found while fetching tables. Need to stop proccessing ")
 		return err
 	}
 
-	var wg sync.WaitGroup
 	// Snapshot sync. Start were we left last
-	for _, tableID := range s.tables {
-		wg.Add(1)
-
-		s.tomb.Go(func() (err error) {
-			sdk.Logger(s.ctx).Trace().Msg(fmt.Sprintf("position %v : %v", tableID, s.getPosition()[tableID]))
-			return s.ReadGoogleRow(rowInput, s.records)
-		})
-		position := s.getPosition()
-		rowInput <- readRowInput{tableID: tableID, offset: position[tableID], positions: position, wg: &wg}
-	}
+	wg.Add(1)
+
+	rowInput := readRowInput{offset: s.position, positions: s.position, wg: &wg}
+	s.tomb.Go(func() (err error) {
+		sdk.Logger(s.ctx).Trace().Msg(fmt.Sprintf("position %v : %v", s.table, s.position))
+		return s.ReadGoogleRow(rowInput, s.records)
+	})
+
 	wg.Wait()
+
 	for {
 		select {
 		case <-s.tomb.Dying():
 			return s.tomb.Err()
 		case <-s.ticker.C:
 			sdk.Logger(s.ctx).Trace().Msg("ticker started ")
-
-			if err = getTables(s); err != nil {
-				sdk.Logger(s.ctx).Trace().Str("err", err.Error()).Msg("error found while fetching tables. Need to stop proccessing ")
-				return err
-			}
 			runCDCIterator(s, rowInput)
 		}
 	}
 }
 
-func runCDCIterator(s *Source, rowInput chan readRowInput) {
+func runCDCIterator(s *Source, rowInput readRowInput) {
 	// wait group make sure that we start new iteration only
 	//  after the first iteration is completely done.
 	var wg sync.WaitGroup
-	for _, tableID := range s.tables {
-		wg.Add(1)
-		offset := s.getPosition()[tableID]
-		position := s.getPosition()
-		s.tomb.Go(func() (err error) {
-			sdk.Logger(s.ctx).Trace().Msg(fmt.Sprintf("position %v : %v", tableID, s.getPosition()[tableID]))
-			return s.ReadGoogleRow(rowInput, s.records)
-		})
-		rowInput <- readRowInput{tableID: tableID, offset: offset, positions: position, wg: &wg}
-	}
+	wg.Add(1)
+	rowInput = readRowInput{tableID: s.table, offset: s.position, positions: s.position, wg: &wg}
```

Comment:
In the ` readRowInput` struct, `offset` and `positions` always have the same value, meaning one of them is redundant.